lib/ramble/ramble/expander.py

```diff
@@ -239,6 +239,14 @@ def workload_namespace(self):
             return '%s.%s' % (app_name, wl_name)
         return None
 
+    @property
+    def spec_namespace(self):
+        app_name = self.expand_var('{spec_name}')
+        wl_name = self.workload_name
+        if app_name and wl_name:
+            return '%s.%s' % (app_name, wl_name)
+        return None
```

Comment:
Do we want to check the returned value at the call sight? I'm not sure if it's actually possible, but a bug where we cache `('spack', None)` as meaning different iterations are the same doesn't sound fun to track down later. 

(given it's based on app and wl name being falsey, that seems unlikely but might be best to be safe?)

Reply:
Thanks!